test/test_filters.rb

```diff
@@ -844,6 +844,11 @@ def to_liquid
         assert_equal 2, @filter.where(@array_of_objects, "color", "red").length
       end
 
+      should "filter objects with null properties appropriately" do
+        array = [{}, { "color" => nil }, { "color" => "" }, { "color" => "text" }]
+        assert_equal 2, @filter.where(array, "color", nil).length
```

Comment:
Why is `2` the value we're expecting here? `nil` and empty strings are not the same thing. Is a liquid thing and not a ruby thing?

Reply:
This is due to code for compatibility with [existing **tests**](https://github.com/jekyll/jekyll/blob/6905c80470d403cb53a40d3e62798b72491903d3/test/test_filters.rb#L940-L955) that check for stringified values.